src/App.tsx

```diff
@@ -1,9 +1,233 @@
+/* eslint-disable max-len */
 import './App.scss';
+import {
+  Routes, Route, Navigate,
+} from 'react-router-dom';
+import { useState, useEffect } from 'react';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header/Header';
+import {
+  ProductsListWithSlider,
+} from './components/Pages/Home/ProductsListWithSlider/ProductsListWithSlider';
+import {
+  ProductsSlider,
+} from './components/Pages/Home/ProductsSlider/ProductsSlider';
+import {
+  ShopByCategory,
+} from './components/Pages/Home/ShopByCategory/ShopByCategory';
+import { CartPage } from './components/Pages/Cart/CartPage';
+import { FavouritesPage } from './components/Pages/Favourites/FavouritesPage';
+import {
+  ProductsCardPage,
+} from
+  './components/Pages/Categories/ProductPagePage/ProductCardList/ProductsCardPage/ProductsCardPage';
+import {
+  ProductDetailsPage,
+} from
+  './components/Pages/Categories/ProductPagePage/ProductCardList/ProductDetailsPage/ProductDetailsPage';
+import { NotFoundPage } from './components/NotFoundPage/NotFoundPage';
+import { Product } from './types/types';
 
-const App = () => (
-  <div className="App">
-    <h1>React Phone Catalog</h1>
-  </div>
-);
+const App = () => {
+  const [IPhones, setIPhones] = useState([]);
+  const [shuffeledIPhones, setShuffeledIPhones] = useState([]);
+  const [visibleIPhones, setVisibleIPhones] = useState<Product[]>(IPhones);
+  const [hotPriceProducts, setHotPriceProducts] = useState([]);
+  const [brandNewProducts, setBrandNewProducts] = useState([]);
+  const [searchInput, setSearchInput] = useState('');
```

Comment:
I suggest to move it to the input component and on submit push the search query into path query so other components will watch the path query and handle changes